bigtable/google/cloud/bigtable/row_data.py

```diff
@@ -274,6 +274,9 @@ def consume_next(self):
 
             self._validate_chunk(chunk)
 
+            if hasattr(self._response_iterator, 'set_start_key'):
+                self._response_iterator.set_start_key(chunk.row_key)
```

Comment:
I think this has to happen after `save_current_row()`, otherwise won't this skip a row if the it fails in the middle?

Reply:
Can you clarify what you mean here? Don't we need to set the start key on any chunk, even if it is not a "commit_row" chunk?

Reply:
This happens in `consume_next()`. I'm worried about a case where a partial row gets retrieved and then an error gets thrown. We set the `start_key` to the row key of the partial chunk that we are working on, but make it open/exclusive. So I think the next read would pick up after the row that we never finished (and would never go back to)? 

I believe we need to wipe out any partial row data on error and make sure we set the start key to the last *completed* row key.

Reply:
Does this work? I moved the check to after rows are commited.